src/test/enhetstester/kotlin/no/nav/familie/ks/sak/kjerne/arbeidsfordeling/ArbeidsfordelingServiceTest.kt

```diff
@@ -123,4 +138,65 @@ internal class ArbeidsfordelingServiceTest {
         verify(exactly = 1) { mockedArbeidsfordelingPåBehandling.copy(0, 0, "testId", "testNavn", true) }
         verify(exactly = 1) { arbeidsfordelingPåBehandlingRepository.save(mockedArbeidsfordelingPåBehandlingEtterEndring) }
     }
+
+    @Nested
+    inner class FastsettBehandlendeEnhet {
+        @Test
+        fun `skal overstyre behandlende enhet fra NORG dersom enhet fra finnArbeidsfordelingForOppgave er en annen`() {
+            // Arrange
+            val behandling = lagBehandling()
+            val søker = lagPerson(personType = PersonType.SØKER, aktør = behandling.fagsak.aktør)
+            val barn = lagPerson(personType = PersonType.BARN, aktør = randomAktør())
+            val arbeidsfordelingsenhet =
+                Arbeidsfordelingsenhet(
+                    enhetId = KontantstøtteEnhet.MIDLERTIDIG_ENHET.enhetsnummer,
+                    enhetNavn = KontantstøtteEnhet.MIDLERTIDIG_ENHET.enhetsnavn,
+                )
+
+            val oppdatertArbeidsfordelingPåBehandling = mockk<ArbeidsfordelingPåBehandling>()
+
+            every {
+                arbeidsfordelingPåBehandlingRepository.finnArbeidsfordelingPåBehandling(behandling.id)
+            } returns null
+
+            every {
+                unleashService.isEnabled(FeatureToggleConfig.OPPRETT_SAK_PÅ_RIKTIG_ENHET_OG_SAKSBEHANDLER, false)
+            } returns true
+
+            every { personopplysningerService.hentPersoninfoEnkel(any()).adressebeskyttelseGradering } returns null
+
+            every {
+                personopplysningGrunnlagRepository
+                    .findByBehandlingAndAktiv(behandling.id)
+            } returns lagPersonopplysningGrunnlag(søkerPersonIdent = søker.aktør.aktivFødselsnummer())
+
+            every { integrasjonClient.hentBehandlendeEnheter(søker.aktør.aktivFødselsnummer()) } returns
+                listOf(
+                    arbeidsfordelingsenhet,
+                )
+
+            every {
+                tilpassArbeidsfordelingService.tilpassArbeidsfordelingsenhetTilSaksbehandler(
+                    arbeidsfordelingsenhet = arbeidsfordelingsenhet,
+                    navIdent = NavIdent(SikkerhetContext.hentSaksbehandler()),
+                )
+            } returns Arbeidsfordelingsenhet(enhetId = KontantstøtteEnhet.OSLO.enhetsnummer, enhetNavn = KontantstøtteEnhet.OSLO.enhetsnavn)
+
+            val arbeidsfordelingPåBehandlingSlot = slot<ArbeidsfordelingPåBehandling>()
+
+            every {
+                arbeidsfordelingPåBehandlingRepository.save(capture(arbeidsfordelingPåBehandlingSlot))
+            } returns oppdatertArbeidsfordelingPåBehandling
```

Comment:
Kan denne erstattes med `returnsArgument 0`? Da slipper man å mocke `oppdatertArbeidsfordelingPåBehandling`

```suggestion
            } returnsArgument 0
```